qualtran/bloqs/block_encoding/chebyshev_polynomial_test.py

```diff
@@ -144,3 +151,80 @@ def test_scaled_chebyshev_odd_tensors():
     bloq = _scaled_chebyshev_poly_odd()
     from_tensors = gate_test(bloq)
     np.testing.assert_allclose(from_gate, from_tensors, atol=1e-14)
+
+
+@frozen
+class TestBlockEncoding(BlockEncoding):
+    """Instance of `BlockEncoding` accepting a matrix with one system, ancilla, and resource bit."""
+
+    matrix: Tuple[Tuple[complex, ...], ...] = field(
+        converter=lambda mat: tuple(tuple(row) for row in mat)
+    )
+    alpha: SymbolicFloat = 1
+    epsilon: SymbolicFloat = 0
+
+    system_bitsize: SymbolicInt = 1
+    ancilla_bitsize: SymbolicInt = 1
+    resource_bitsize: SymbolicInt = 1
+
+    @cached_property
+    def signature(self) -> Signature:
+        return Signature.build(system=1, ancilla=1, resource=1)
+
+    @property
+    def signal_state(self) -> PrepareOracle:
+        raise NotImplementedError
+
+    @property
+    def target_registers(self) -> Tuple[Register, ...]:
+        raise NotImplementedError
+
+    @property
+    def junk_registers(self) -> Tuple[Register, ...]:
+        raise NotImplementedError
+
+    @property
+    def selection_registers(self) -> Tuple[Register, ...]:
+        raise NotImplementedError
+
+    def build_composite_bloq(
+        self, bb: BloqBuilder, system: Soquet, ancilla: Soquet, resource: Soquet
+    ) -> Dict[str, SoquetT]:
+        bits = bb.join(np.concatenate([bb.split(system), bb.split(ancilla), bb.split(resource)]))
+        bits = bb.add(MatrixGate(3, self.matrix, atol=3e-8), q=bits)
+        system, ancilla, resource = bb.split(bits)
+        return {"system": system, "ancilla": ancilla, "resource": resource}
+
+
+def block_encoding_matrix(a):
```

Comment:
perhaps this should be a factory constructor for `TestBlockEncoding` as the behaviour seems tightly coupled (for now)